vyxal/parse.py

```diff
@@ -270,58 +159,87 @@ def parse(tokens: list[lexer.Token]) -> list[Structure]:
                                     lambdas are just normal lambdas +
                                     an element.
             """
-            after_token = None
-            if structure_name == StructureType.FOR_LOOP:
+            if structure_cls == structure.ForLoop:
                 if len(branches) > 1:
-                    branches[0] = variable_name(branches[0])
-                branches[-1] = parse(branches[-1])
-            elif structure_name == StructureType.FUNCTION:
-                print(branches[0])
-                branches[0] = process_parameters(branches[0])
+                    var_names = [
+                        variable_name(branch) for branch in branches[:-1]
+                    ]
+                body = parse(branches[-1])
+                structures.append(structure.ForLoop(var_names, body))
+            elif structure_cls == structure.WhileLoop:
+                if len(branches) == 1:
+                    # If there's no condition, it's an infinite loop
+                    condition = [lexer.Token(lexer.TokenType.NUMBER, "1")]
```

Comment:
this seems like an overly complicated way of handling an infinite loop. Is there some more elegant way of setting the condition to `1` (or `True`)?

Reply:
At least with this way, `1` is always placed inside the context variable. I prefer this behaviour.